### PR TITLE
[DEV-6532] About the Data Code Clean Up pt1 (Renaming Redux Properties & Related Variable/Fn Names)

### PR DESCRIPTION
**Pull request by MaxwellKendall**
**High level description:**
- Updating the redux property names
- Updating local state / variables that refer to redux values
- Updating models and tests

**Technical details:**

f845894 & 6bc847f - model updates resolving minor errors and reusing code
482459c - renaming a lot of stuff

**JIRA Ticket:**
[DEV-6532](https://federal-spending-transparency.atlassian.net/browse/DEV-6532)

The following are ALL required for the PR to be merged:

Author:
- [x] Linked to this PR in JIRA ticket
`N/A` Scheduled demo including Design/Testing/Front-end OR Provided instructions for testing in JIRA and PR `if applicable`
`N/A` Verified cross-browser compatibility: Chrome, Safari, Firefox, Internet Explorer, Edge
`N/A` Verified mobile/tablet/desktop/monitor responsiveness
`N/A`Verified that this PR does not create any *new* accessibility issues (via Axe Chrome extension)
- [x] Added Unit Tests for helper functions, reducers, models and Container/Component Interactivity Expectations `if applicable` [React Testing Library](react-testing-library.md)
`N/A` [API contract](https://github.com/fedspendingtransparency/usaspending-api/tree/dev/usaspending_api/api_contracts) updated `if applicable`
`N/A` [Component Library Integration Status Report](https://github.com/fedspendingtransparency/data-act-documentation/blob/data-transparency-ui/frontend_apps/component-library-integration-status.md) updated `if applicable`

Reviewer(s):
`N/A` Design review complete `if applicable`
`N/A` [API #1234](https://github.com/fedspendingtransparency/usaspending-api/pull/1234) merged concurrently `if applicable`
- [ ] Code review complete

### CODE REVIEW

File: src/js/containers/aboutTheData/AgenciesContainer.jsx [hunk 1]
```diff
@@ -145,13 +145,13 @@ const AgenciesContainer = ({
     });
 
     useEffect(() => {
-        if (datesReq.current) {
+        if (publicationsReq.current) {
             console.info('canceling request on unmount');
```
Comment (jonhill13) [inline src/js/containers/aboutTheData/AgenciesContainer.jsx:149]: shouldn't this say `canceling request on mount`?

Unmount would be 
```
useEffect(() => {
  return () => {
    // cancel all the requests
  };
});
```

File: src/js/containers/aboutTheData/AgenciesContainer.jsx [hunk 2]
```diff
@@ -145,13 +145,13 @@ const AgenciesContainer = ({
...
-            datesReq.current.cancel();
+            publicationsReq.current.cancel();
         }
-        if (detailsReq.current) {
+        if (submissionsReq.current) {
             console.info('canceling request on unmount');
```
Comment (jonhill13) [inline src/js/containers/aboutTheData/AgenciesContainer.jsx:153]: same as above

File: src/js/containers/aboutTheData/AgenciesContainer.jsx [hunk 3]
```diff
@@ -145,13 +145,13 @@ const AgenciesContainer = ({
...
-            detailsReq.current.cancel();
+            submissionsReq.current.cancel();
         }
         if (totalsReq.current) {
             console.info('canceling request on unmount');
```
Comment (jonhill13) [inline src/js/containers/aboutTheData/AgenciesContainer.jsx:157]: same as above

File: src/js/containers/aboutTheData/AgenciesContainer.jsx [hunk 4]
```diff
@@ -210,62 +210,62 @@ const AgenciesContainer = ({
...
-                            direction: datesSort[1]
+                            field: publicationsSort[0],
+                            direction: publicationsSort[1]
                         }}
                         error={error}
                         loading={areDatesLoading} />
                 )}
             </div>
-            {(details.length || dates.length) && (
+            {(allSubmissions.length || allPublications.length) && (
```
Comment (jonhill13) [inline src/js/containers/aboutTheData/AgenciesContainer.jsx:260]: Do we want to show pagination on error and loading?

**Review by jonhill13 — CHANGES_REQUESTED** [review]

File: src/js/containers/aboutTheData/AgenciesContainer.jsx (hunk 4 above)
Comment (MaxwellKendall) [reply src/js/containers/aboutTheData/AgenciesContainer.jsx:260]: @jonhill13 https://github.com/fedspendingtransparency/usaspending-website/pull/2302/files?file-filters%5B%5D=.jsx#diff-5e7810fb3a893bd1fbfd8f908507a8a1ebd31de25d46717b5bb3273a992846d0R265

File: src/js/containers/aboutTheData/AgenciesContainer.jsx [hunk 5]
```diff
@@ -145,13 +145,13 @@ const AgenciesContainer = ({
     });
 
     useEffect(() => {
```
Comment (MaxwellKendall) [inline src/js/containers/aboutTheData/AgenciesContainer.jsx:147]:
```suggestion
    useEffect(() => () => {
```

**Review by jonhill13 — APPROVED** [review]

**Pull request merged**